core/unit_test/cuda/TestCuda_TeamSizeMaxIsPositive.cpp

```diff
@@ -0,0 +1,111 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestCuda_Category.hpp>
+#include <Kokkos_Core.hpp>
+
+namespace Test {
+namespace Impl {
+
+struct FunctorFor {
+  KOKKOS_FUNCTION
+  void operator()(
+      Kokkos::TeamPolicy<TEST_EXECSPACE>::member_type const&) const {}
+};
+
+struct TeamSizeTester {
+  void run_test_team_size(Kokkos::AUTO_t teamSizeRequested,
+                          Kokkos::AUTO_t vectorLengthRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested,
+                                              vectorLengthRequested);
```

Comment:
I am struggling to see why we define 6 overloads when it feels like at the end of the day we just want to forward to the team policy constructor and then the body of the test is essentially the same.  Why is it not one single variadic overload?